code/game/objects/structures/barricade/barricade.dm

```diff
@@ -35,6 +42,8 @@
 	var/burn_flame_multiplier = 1
 	var/repair_materials = list()
 	var/metallic = TRUE
+	/// var that say what state a barricade can't be fix by welder. null mean it can be repair no matter the state.
+	var/welder_fix = null
```

Comment:
Could we name this something more descriptive like `welder_lower_damage_limit` or something then we should be good to go,